adapter/validator/src/main/java/org/mobilitydata/gtfsvalidator/validator/GtfsTypeValidator.java

```diff
@@ -95,39 +99,84 @@ public GtfsTypeValidator(@NotNull GtfsSpecificationProto.CsvSpecProto fileSchema
+                        }
+                        break;
+                    }
+                    case LONGITUDE: {
+                        if (!floatValidator.isInRange(
+                                (Float) value,
+                                LONGITUDE_MIN_VALUE,
+                                LONGITUDE_MAX_VALUE)) {
+
+                            toReturn.add(new FloatFieldValueOutOfRangeNotice(
+                                    toValidate.getRawFileInfo().getFilename(),
+                                    columnSpecProto.getName(),
+                                    toValidate.getEntityId(),
+                                    LONGITUDE_MIN_VALUE,
+                                    LONGITUDE_MAX_VALUE,
+                                    (Float) value)
+                            );
                         }
                         break;
                     }
-                    case INTEGER: {
+                    case NON_NEGATIVE_FLOAT: {
+                        if (!floatValidator.isInRange(
+                                (Float) value,
+                                0,
+                                Float.MAX_VALUE)) {
+
+                            toReturn.add(new FloatFieldValueOutOfRangeNotice(
+                                    toValidate.getRawFileInfo().getFilename(),
+                                    columnSpecProto.getName(),
+                                    toValidate.getEntityId(),
+                                    0,
+                                    Float.MAX_VALUE,
+                                    (Float) value)
+                            );
+                        }
+                        break;
+                    }
+                    case NON_NEGATIVE_INTEGER: {
                         if (!integerValidator.isInRange(
                                 (Integer) value,
-                                columnSpecProto.getIntmin(),
-                                columnSpecProto.getIntmax())) {
+                                0,
+                                Integer.MAX_VALUE)) {
 
                             toReturn.add(new IntegerFieldValueOutOfRangeNotice(
                                     toValidate.getRawFileInfo().getFilename(),
                                     columnSpecProto.getName(),
                                     toValidate.getEntityId(),
-                                    columnSpecProto.getIntmin(),
-                                    columnSpecProto.getIntmax(),
-                                    (Integer) value
-                            ));
+                                    0,
+                                    Integer.MAX_VALUE,
+                                    (Integer) value)
+                            );
+                        }
+                        break;
+                    }
+                    case NON_NULL_INTEGER: { // a null integer is an integer with value=0
```

Comment:
@LeoFrachet the root of the confusion. I am not certain we came to a satisfyingly clear solution. I'll think more overt this.